tests/test_cargo_build_lib.rs

```diff
@@ -53,6 +53,32 @@ test!(build_with_no_lib {
                        .with_stderr("no library targets found"));
 });
 
+test!(build_global_lock {
+    
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [package]
+
+            name = "foo"
+            version = "0.0.1"
+            authors = ["wycats@example.com"]
+        "#)
+        .file("src/main.rs", r#"
+            fn main() {}
+        "#)
+        .file("src/lib.rs", r#" "#);
+
+    p.build();
+    
+    let mut bg_process = p.cargo("build");
+    let fg_process = bg_process.clone();
+    let mut bg_process_handle = bg_process.build_command().spawn().unwrap();
+
+    assert_eq!(fg_process.build_command().output().unwrap().status.code().unwrap(), 101);
```

Comment:
Yes, this is also why it doesn't work most of the time. I am not sure how to make it work without having to make the two processes communicate (i.e. master waits for child to come up).

Maybe I could launch the cargo implementation from within the test-case directly (without a sub-process), and then spawn the test-executable. That would give me more control, and is similar to the [working tests](https://github.com/Byron/file-lock/blob/cargo-adjustments/tests/process_filename_lock.rs#L43)  in the `file-lock` crate.